### PR TITLE
Configure set-cookie header

### PR DESCRIPTION
**Pull request by spenes**
This PR brings necessary changes to configure set-cookie header.

### CODE REVIEW

File: http4s/src/main/scala/com.snowplowanalytics.snowplow.collectors.scalastream/model.scala [hunk 1]
```diff
@@ -27,7 +29,21 @@ object model {
     */
   final case class SplitBatchResult(goodBatches: List[List[Json]], failedBigEvents: List[Json])
 
-  final case class CollectorConfig(
-    paths: Map[String, String]
+  final case class CookieConfig(
+    enabled: Boolean,
+    name: String,
+    expiration: FiniteDuration,
+    domains: Option[List[String]],
```
Comment (istreeter) [inline http4s/src/main/scala/com.snowplowanalytics.snowplow.collectors.scalastream/model.scala:36]: Is there any significant difference between `None` and `Some(Nil)`?  Did you consider making this a non-optional `List[String]` with a default of `Nil`?

**Review by istreeter — COMMENTED** [review]
👍 Nice to see an important part of the collector functionality in the http4s project.

Just a couple of questions to think about.

File: http4s/src/main/scala/com.snowplowanalytics.snowplow.collectors.scalastream/CollectorService.scala [hunk 2]
```diff
@@ -152,6 +166,100 @@ class CollectorService[F[_]: Sync](
...
+        case Some(_) => None
+        case None =>
+          cookieConfig.map { config =>
+            val responseCookie = ResponseCookie(
+              name    = config.name,
+              content = networkUserId,
+              expires = Some(HttpDate.unsafeFromEpochSecond((now + config.expiration).toSeconds)),
+              domain  = cookieDomain(headers, config.domains, config.fallbackDomain),
+              path    = Some("/"),
+              sameSite = config.sameSite.flatMap {
```
[the remark this answers is not in the document]
Comment (spenes) [reply http4s/src/main/scala/com.snowplowanalytics.snowplow.collectors.scalastream/CollectorService.scala:198]: Yep, you are right. It should be easy thing to do. I will change that.

File: http4s/src/main/scala/com.snowplowanalytics.snowplow.collectors.scalastream/model.scala (hunk 1 above)
Comment (spenes) [reply http4s/src/main/scala/com.snowplowanalytics.snowplow.collectors.scalastream/model.scala:36]: I don't think there is a difference between `None` and `Some(Nil)`. We can use `List[String]` without wrapping it to `option` I guess. I will make necessary changes.

**Review by istreeter — APPROVED** [review]

**Pull request merged**